### DISCUSSION TITLE
Use a different filesystem URL for PDF files

### DISCUSSION THREAD

**Discussion by mgtcampbell** [statamic/cms#6041] [Q&A]
Hi there,

I currently use AWS + imgix to serve up image assets through our websites, with the imgix url used to then serve up those assets on the front end.

The issue is that if a client then uploads a PDF file through the AWS assets container, by default it then uses the imgix url and will only display the first page of the PDF due to imgix's processing.

Is there an easy way to add a conditional in the filesystem config so that if the file is _not_ an image to use the s3 url instead of the imgix one?

Cheers

**Comment by jasonvarga** [comment]
How are you configuring the imgix url?

**Reply by mgtcampbell** [reply]
In the filesystems.php config file:

```
'assets_aws_s3' => [
            'driver' => 's3',
            'key' => env('AWS_ACCESS_KEY_ID'),
            'secret' => env('AWS_SECRET_ACCESS_KEY'),
            'region' => env('AWS_DEFAULT_REGION'),
            'bucket' => env('AWS_BUCKET'),
            'url' => env('AWS_URL'),
        ],
```

So the AWS_URL environment variable is set to `https://example.imgix.net`

I guess I could set that to the s3 url instead, and then only serve up my image partials with the imgix url, or create a separate bucket for files and instruct the client to upload "files" like PDFs there... was just looking for something a bit neater since it's only for a few isolated PDF files.

Cheers
Mark

**Reply by jasonvarga** [reply]
Yeah I'd keep the container URL as the "real" s3 url, then manually use imgix where you need it.

**Reply by mgtcampbell** [reply]
Cool - thanks for your advice and all the awesome work you guys do
